store/pizza-slice.ts

```diff
@@ -1,43 +1,119 @@
 import { createSlice, PayloadAction } from "@reduxjs/toolkit";
 import { RootState } from "store";
-import { IPizza, OnAddPizza, OnAddCalculate } from 'types'
+import { IPizza, addPizza, uniquePizza, Count, IncrementDecrement } from 'types'
 
 
 const initialState = {
   list: [] as IPizza[],
-  selected: {} as OnAddPizza,
-  calculated: {} as OnAddCalculate
+  products: [] as addPizza[],
+  uniqueProductsList: [] as uniquePizza[],
+  productCount: 0,
+  totalPrice: 0,
 }
 
 const slice = createSlice({
   name: 'PIZZA',
   initialState,
   reducers: {
-    onSave: (state, action: PayloadAction<IPizza[] | never[]>) => {
-        state.list = action.payload 
+    onSave: (state, action: PayloadAction<IPizza[]>) => {
+      state.list = action.payload 
     },
-    onAddPizza: (state, action: PayloadAction<OnAddPizza >) => {
-      const {id, pizza} = action.payload
-      if (!(id in state.selected)) state.selected[id] = [pizza]
-      else state.selected[id].push(pizza)
+    setProducts: (state, action: PayloadAction<addPizza >) => {
+      const product = action.payload;
+      state.products.push(product);
+
+      state.productCount = state.products.length;
+      state.totalPrice += action.payload.price;
+    },
+    
+    onVariantCount: (state, action: PayloadAction<addPizza[]>) => {
```

Comment:
`addPizza` isn't appropriate type name, more sounds like function name. Please rename it :)